object-manipulation-2/src/omit.js

```diff
@@ -0,0 +1,16 @@
+/* exported omit */
+// Make storage for the output
+// check each value of the array and see if it matches the properties of the source.
+// if a property matches an array value, omit the properties and store the remain propeties into storage
+// check for edge cases such as null, undefined, or empty object literals.
+function omit(source, key) {
+  var objectOutput = {};
+  for (var i = 0; i < key.length; i++) {
+    for (var prop in source) {
+      if (prop === key[i] || prop === key[i + 1] || prop === key[0]) {
+        objectOutput;
```

Comment:
What is the point of this code block? The work is just saying the name of a variable and doing nothing with it, this does nothing. What was your logic here?

Reply:
It looks like the actual work you want to do here is in your `else` block. What are you _actually_ searching for here?

Reply:
My logic was that if any of the keys match with property names instead of deleting the properties as  they turn undefined and gave error i would simply do nothing, unfortunately i dont know how to do nothing without returning the function, and whatever does not fit those requirments it puts those in the new output. I kept getting more outputs than necessary so that was my guard check because it looped twice once from the property name checking at key[i] for every i.